### PR TITLE
Fix InternVL2-8B compiled generation with InternLM2 backbone

### LINKED ISSUE #43631: [Bug]: enforce_eager=False will generate garbage tokens for InternVL2-8B
### Your current environment

<details>

<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```
==============================
        System Info
==============================
OS                           : Ubuntu 22.04.5 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version                : 14.0.0-1ubuntu1.1
CMake version                : version 3.22.1
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.11.0+cu130
Is debug build               : False
CUDA used to build PyTorch   : 13.0
ROCM used to build PyTorch   : N/A
XPU used to build PyTorch    : N/A

==============================
      Python Environment
==============================
Python version               : 3.10.12 (main, Mar  3 2026, 11:56:32) [GCC 11.4.0] (64-bit runtime)
Python platform              : Linux-6.8.0-110-generic-x86_64-with-glibc2.35
    
==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : Could not collect
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : 
GPU 0: NVIDIA GeForce RTX 5090
GPU 1: NVIDIA GeForce RTX 5090

Nvidia driver version        : 580.82.07
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True
</details>

## Summary

When running vLLM with the default compiled mode (`enforce_eager=False`) on an RTX 5090, the model produces incorrect/repetitive garbage tokens.
Switching to `enforce_eager=True` immediately restores correct output.
No exception, no NaN, no warning is raised — the failure is completely silent.
---

## Reproduction

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "1"

from vllm import LLM, SamplingParams

messages = [{"role": "user", "content": "What is the capital of France?"}]

# BROKEN — compiled mode (default)
llm = LLM(model="OpenGVLab/InternVL2-8B", trust_remote_code=True)
out = llm.chat(messages, SamplingParams(temperature=0, max_tokens=50))
print(out[0].outputs[0].text)
# → "down down down down down down down down down down ..."

# WORKS — eager mode
llm = LLM(model="OpenGVLab/InternVL2-8B", trust_remote_code=True, enforce_eager=True)
out = llm.chat(messages, SamplingParams(temperature=0, max_tokens=50))
print(out[0].outputs[0].text)
# → "The capital of France is Paris."
```

Tested with a **cold cache** (cache deleted before each run) — the issue is not caused
by a stale compiled artifact. The freshly compiled binary itself is incorrect.
## Observed vs Expected

| | Output |
|---|---|
| **Compiled mode** (`enforce_eager=False`) | `"down down down down down down..."` |
| **Eager mode** (`enforce_eager=True`) | `"The capital of France is Paris."` |

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by YuanaHao**
Fixes #43631.

InternVL feeds precomputed text+vision embeddings into the text backbone. For InternVL2 models whose text backbone is `InternLM2ForCausalLM`, the compiled internal InternLM2 model can produce corrupted logits on this `inputs_embeds` path while eager execution is correct.

This PR disables torch.compile only for the internal InternLM2 text backbone when used inside `InternVLChatModel`. CUDA graph capture and the rest of the model behavior remain enabled.

Validation run:
- `uv run --no-project --with ruff ruff format --check vllm/model_executor/models/internvl.py tests/models/multimodal/processing/test_internvl.py`
- `uv run --no-project --with ruff ruff check vllm/model_executor/models/internvl.py tests/models/multimodal/processing/test_internvl.py`
- `python -m pytest -q tests/models/multimodal/processing/test_internvl.py::test_internvl_internlm2_backbone_disables_torch_compile`
- E2E on H20 with `OpenGVLab/InternVL2-8B`, default `enforce_eager=False`: `The capital of France is Paris.`

### CODE REVIEW

**Comment by YuanaHao** [comment]
Updated after the automated review:

- Guarded the inner language model access with `getattr(self.language_model, "model", None)` before setting `do_not_compile`.
- Amended the commit with the required DCO `Signed-off-by` line.
- Re-ran local ruff format/check, the targeted unit test, and the InternVL2-8B E2E repro on the H20 node.
